### ISSUE TITLE
Rolling resatrt only possible when cluster is green

### ISSUE THREAD

**Issue by saketmht** [Opster/opensearch-k8s-operator#643]
The operator implements rolling restart for the pods i.e. if there are any changes, the operator will manually restart the pods so that the latest changes are reflected.

Before we do attempt to restart, we are checking the cluster health to be green and if it is not then we don't proceed with the restart. But in some cases, it might be such that the cluster health can only be fixed via some changes to the CR and to the pods but since we don't all restarting without green health, the operator is stuck forever to wait for green health and the new updates are never picked up.

I think the operator should check for the availability of certain number of nodes (maybe data nodes) rather than checking for health or maybe a flag to allow restart even with yellow health.

**Comment by dbason** [comment]
Where there is no persistent storage this could potentially lead to data loss.  Perhaps there could be a field where we allow a user to choose thresholds for continuing with a restart.

**Comment by synhershko** [comment]
This is by design. Cluster should definitely be green before starting a rolling restart. We can maybe allow to bypass that via specific config to allow yellow-state restarts but let's have this conversation on a new issue if anyone is still interested in that
